File: server/parsing/grouped-results-parser/src/main/java/projektor/parser/grouped/model/GroupedResults.java
package projektor.parser.grouped.model;

import java.math.BigDecimal;
import java.util.List;

public class GroupedResults {
    public List<GroupedTestSuites> groupedTestSuites;
    public List<PerformanceResult> performanceResults;
    public ResultsMetadata metadata;
    public BigDecimal wallClockDuration;
    public List<CoverageFile> coverageFiles;
    public List<CodeQualityReport> codeQualityFiles;
}

File: server/server-app/src/main/kotlin/projektor/incomingresults/model/GroupedResults.kt
package projektor.incomingresults.model

import projektor.parser.coverage.payload.CoverageFilePayload
import projektor.server.api.quality.CodeQualityReport
import java.math.BigDecimal

data class GroupedResults(
    val groupedTestSuites: List<GroupedTestSuites>,
    val performanceResults: List<PerformanceResult>,
    val metadata: ResultsMetadata?,
    val wallClockDuration: BigDecimal?,
    val coverageFiles: List<CoverageFilePayload>?,
    val codeQualityReports: List<CodeQualityReport>?
)
